recipes/libgit2/meta.yaml

```diff
@@ -28,6 +28,8 @@ requirements:
     - cmake
     - python  # [win]
     - msinttypes  # [win]
+    - openssl     # [not win]
```

Comment:
Only reason this for the switch is because on the [website](https://github.com/libgit2/libgit2#optional-dependencies), it is mentioned as non-Windows dependency. Should we keep the switch as is or remove it?

> OpenSSL (non-Windows) to talk over HTTPS and provide the SHA-1 functions